File: core/client/game/src/commonGame/scene/role/RoleAttributeLogic.cs
using System;
using ShineEngine;

/// <summary>
/// 角色属性逻辑
/// </summary>
public class RoleAttributeLogic:RoleLogicBase
{
	/** 属性模块 */
	protected RoleAttributeDataLogic _aTool;

	private MUnitAttributesChangeEventObj _attributeEventObj=new MUnitAttributesChangeEventObj();

	public override void construct()
	{
		base.construct();

		_aTool=new RoleAttributeDataLogic(this);
	}

	public override void init()
	{
		base.init();

		_aTool.setData(_data.attribute.attributes);
	}

	public override void onFrame(int delay)
	{
		base.onFrame(delay);

		_aTool.onPiece(delay);
	}

	/** 获取属性逻辑 */
	public RoleAttributeDataLogic getAttribute()
	{
		return _aTool;
	}

	/** 属性改变 */
	public virtual void onAttributeChange(int[] changeList,int num,bool[] changeSet,int[] lastAttributes)
	{
		Ctrl.print("BB",_role.playerID,GameC.player.role.playerID);

		if(_role.isSelf())
		{
			Ctrl.print("CC");

			_attributeEventObj.index=0;
			_attributeEventObj.changeSet=changeSet;

			GameC.player.dispatch(GameEventType.MRoleAttributeChange,_attributeEventObj);
		}
	}
}

File: core/server/project/commonBase/src/main/java/com/home/commonBase/scene/role/RoleAttributeLogic.java
package com.home.commonBase.scene.role;

import com.home.commonBase.logic.role.RoleAttributeDataLogic;
import com.home.commonBase.scene.base.RoleLogicBase;
import com.home.shine.support.collection.IntIntMap;

/** 玩家属性逻辑 */
public class RoleAttributeLogic extends RoleLogicBase
{
	/** 属性模块 */
	protected RoleAttributeDataLogic _aTool;
	
	@Override
	public void construct()
	{
		super.construct();
		
		_aTool=new RoleAttributeDataLogic(this);
	}
	
	@Override
	public void init()
	{
		super.init();
		
		_aTool.setData(_data.attribute.attributes);
	}
	
	@Override
	public void onFrame(int delay)
	{
		super.onFrame(delay);
		
		_aTool.onPiece(delay);
	}
	
	/** 获取属性逻辑 */
	public RoleAttributeDataLogic getAttribute()
	{
		return _aTool;
	}
	
	/** 推送自己属性 */
	public void sendSelfAttribute(IntIntMap dic)
	{
	
	}
	
	/** 推送别人属性 */
	public void sendOtherAttribute(IntIntMap dic)
	{
	
	}
	
	/** 属性改变 */
	public void onAttributeChange(int[] changeList,int num,boolean[] changeSet,int[] lastAttributes)
	{
	
	}
}
